packages/airnode-deployer/src/infrastructure/aws.ts

```diff
@@ -31,15 +32,48 @@ export const getAirnodeBucket = async () => {
     throw new Error(`Multiple Airnode buckets found, stopping. Buckets: ${JSON.stringify(airnodeBuckets)}`);
   }
 
-  return airnodeBuckets?.[0]?.Name ?? null;
+  const bucketName = airnodeBuckets?.[0]?.Name;
+  if (!bucketName) {
+    logger.debug('No Airnode S3 bucket found');
+    return null;
+  }
+
+  const goBucketLocation = await go(() => s3.getBucketLocation({ Bucket: bucketName }).promise());
+  if (!goBucketLocation.success) {
+    throw new Error(`Failed to get location for bucket '${bucketName}': ${goBucketLocation.error}`);
+  }
+
+  let region = goBucketLocation.data.LocationConstraint;
+  // The `EU` option is listed as a possible one in the documentation
+  // https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#getBucketLocation-property
+  if (isNil(region) || region === 'EU') {
+    throw new Error(`Unknown bucket region '${region}'`);
+  }
+  // The documentation says that for buckets in the `us-east-1` region the value of `LocationConstraint` is null but it is actually an empty string...
+  // https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#getBucketLocation-property
+  if (region === '') {
```

Comment:
🙈

Reply:
:man_shrugging: I'm not happy about it either. Take it up with the AWS SDK developers :smile: